aten/src/ATen/cuda/LegacyTHFunctionsCUDA.cpp

```diff
@@ -465,10 +465,12 @@ std::tuple<Tensor,Tensor> _th_mode(const Tensor & self, int64_t dim, bool keepdi
     }
     return std::tuple<Tensor, Tensor>(values, indices);
 }
-std::tuple<Tensor &,Tensor &> _th_sort_out(Tensor & values, Tensor & indices, const Tensor & self, int64_t dim, bool descending) {
+std::tuple<Tensor &,Tensor &> _th_sort_out_stable(Tensor & values, Tensor & indices, const Tensor & self, c10::optional<bool> stable, int64_t dim, bool descending) {
     // DeviceGuard omitted
     auto dispatch_scalar_type = infer_scalar_type(self);
 
+    TORCH_CHECK(!stable.value(), "stable=True is not implemented on CUDA yet.");
```

Comment:
This check doesn't match the error message. What if, for example, the stable optional has a value and it's False?

Reply:
Yes, always.

Reply:
Also, when it comes to making `sort` stable for CUDA, it is actually non-trivial and requires some research. However, using the same technique as for the CPU port, it is possible to employ `thrust` for this task.

Reply:
Thank you the explanation. Would you add an assertion that the optional has a value (since it's an assumption) and file a follow-up issue (and link it in a comment in the code) for supporting stable sort on CUDA, too?